src/router.js

```diff
@@ -1,41 +1,38 @@
-export function router(options) {
-  return function(emit) {
-    return {
-      state: {
-        router: {}
-      },
-      actions: {
-        router: {
-          set: function(state, actions, data) {
-            return {
-              router: data
-            }
+import { h } from 'hyperapp'
+
+export function router(routes) {
+  return function (app) {
+    return function (props) {
+      return app(enhance(props))
+
+      function enhance(props) {
+        props = Object.assign({ state: {}, actions: {} }, props)
+
+        props.state.router = {}
+        props.actions.router = {
+          set: function (state, actions, data) {
+            return data
           },
-          go: function(state, actions, path) {
+          go: function (state, actions, path) {
             if (location.pathname + location.search !== path) {
-              history.pushState({}, "", path)
-              actions.router.set({
-                path: path
-              })
+              history.pushState({}, '', path)
+              actions.set({ path: path })
             }
           }
         }
-      },
-      events: {
-        load: function(state, actions) {
-          addEventListener("popstate", function() {
+
+        props.view = function (state, actions) {
```

Comment:
This seems like a potentially dangerous pattern with HOA. What happens if a view was defined in the existing app chain?

Reply:
events gone, so the only possibly entrypoint is the view function, but i think that is ok. the previous implementation of the router was doing pretty much the same on the `render` event.

Reply:
please see my lates commit, that is overwriting the default view only when the route matches